docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,559 @@
+
+## How to debug transformations
+
+The most popular tool for transformations debugging is `ngraph::pass::VisualizeTree` transformation that visualize ngraph::Function.
+
+Usage example:
+
+~~~~~~~~~~~~~{.cpp}
+std::vector<std::shared_ptr<ngraph::Function> > g{f};
+// Serialize ngraph::Function to before.svg file before transformation
+ngraph::pass::VisualizeTree("/path/to/file/before.svg").run_on_module(g);
+// Run your transformation
+ngraph::pass::MyTransformation().run_on_function();
+// Serialize ngraph::Function to after.svg file after transformation
+ngraph::pass::VisualizeTree("/path/to/file/after.svg").run_on_module(g);
+~~~~~~~~~~~~~
+
+ngraph::pass::VisualizeTree can be parametrized via environment variables:
+
+`NGRAPH_VISUALIZE_TREE_OUTPUT_SHAPES=1` - visualize shapes
+
+`NGRAPH_VISUALIZE_TREE_OUTPUT_TYPES=1`  - visualize types
+
+Note: current VisualTree has not user friendly interface and it will be changed in nearest future. The intention is to move visualize abilities inside transformations.
+
+If you are using `ngraph::pass::Manager` to run sequence of transformations you can get additional debug capabilities by using next environment variables:
+
+`NGRAPH_PROFILE_PASS_ENABLE=1` - enables performance measurement for each transformation and prints execution status
+
+`NGRAPH_ENABLE_VISUALIZE_TRACING=1` -  enables visualization after each transformation. By default it saves dot and svg files.
+
+Note: make sure that you have dot installed on your machine otherwise it will silently save only dot file without svg file.
+
+## Disabling/Enabling specific transformations for plugin X	 
+
+This topic mostly related to conversion to legacy opset and plugins that based on CNNNetwork but still this mechanism can be applied for other cases.
+Let's suppose that plugin X starts support `opset3::StridedSlice` operation and you want to disable `ConvertStridedSliceToCrop` transformation for plugin X.
+To do this you need to extend transformation class with `ngraph::pass::PassParam` class. This class extends transformations class with `transformation_callback` that can be set by plugin that uses legacy conversion. 
+
+~~~~~~~~~~~~~{.cpp}
+// Extend transformation class with PassParam
+class ngraph::pass::ConvertStridedSliceToCrop: public ngraph::pass::GraphRewrite, public ngraph::pass::PassParam {
+    ...
+}
+
+// Update callback to be able to use transformation_callback if this transformation based on GraphRewrite.
+ngraph::graph_rewrite_callback callback = [this](pattern::Matcher &m) {
+    ...
+}
+
+// Use transformation_callback not to execute transformation
+if (transformation_callback(node)) {
+    return false;
+}
+~~~~~~~~~~~~~
+
+TODO: link to existing example
+
+## Transformations testing
+
+We have two types of tests: nGraph reader tests located in `inference-engine/tests/functional/inference_engine/ngraph_reader` and transformation tests located  in `inference-engine/tests/functional/inference_engine/transformations`
```

Comment:
double space between `located` and `in` ?

Reply:
done.